### PR TITLE
Clean up `roadway_objects` package

### LINKED ISSUE #2145: Clean up `roadway_objects` package
### Summary

The `roadway_objects` package uses a non-standard and somewhat old style of CMake configurations. The formatting should be cleaned up. Additionally, the package should be run through the common ROS 2 linters to improve readability and code quality.

### Reasoning for new functionality

_No response_

### PR DESCRIPTION
**Pull request by adamlm**
# PR Details
## Description

This PR cleans up the `roadway_objects` package by modernizing the CMake configuration and adding the common ROS linters. The code has been refactored to pass linter requirements.

## Related GitHub Issue

Closes #2145 

## Related Jira Key

CDAR-186

## Motivation and Context

The existing `roadway_objects` package has some coding style issues that should be addressed. Since this package needs to be worked on as part of CDAR-186, cleaning it up before other refactoring/enhancement work is a logical first step.

## How Has This Been Tested?

No functionality changes. Existing tests pass.

## Types of changes

- [x] Refactoring/clean up

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [x] All new and existing tests passed.

### CODE REVIEW

File: roadway_objects/src/roadway_objects_worker.cpp [hunk 1]
```diff
@@ -1,56 +1,61 @@
...
+    RCLCPP_WARN_STREAM(
+      logger_->get_logger(),
+      "roadway_objects could not process external objects as no semantic map was available");
     return;
   }
 
-  if (map->laneletLayer.size() == 0)
-  {
-    RCLCPP_WARN_STREAM(logger_->get_logger(), "roadway_objects could not process external objects as the semantic map does not contain any lanelets");
+  if (map->laneletLayer.size() == 0) {
```
Comment (adev4a) [inline roadway_objects/src/roadway_objects_worker.cpp:38]: Since we are refactoring, can we change this check to .empty()?

**Review by adev4a — COMMENTED** [review]
This looks pretty good to me, just left one minor comment. Also, since we are just refactoring this package, can we change the copyright to 2019-2023?

File: roadway_objects/src/roadway_objects_worker.cpp (hunk 1 above)
Comment (adamlm) [reply roadway_objects/src/roadway_objects_worker.cpp:38]: The `RoadwayObjectsWorker` is going to be removed when #2144 is resolved, so I'm inclined to keep that line as is. What are your thoughts?

**Comment by adamlm** [comment]
> This looks pretty good to me, just left one minor comment. Also, since we are just refactoring this package, can we change the copyrights to 2019-2023?

@adev4a There's more work that's going to be done to this package. Can we push that off to a future PR?

**Review by adev4a — APPROVED** [review]

**Pull request merged**